corehq/apps/case_importer/static/case_importer/js/import_history.js

```diff
@@ -19,7 +19,25 @@ hqDefine('case_importer/js/import_history.js', function () {
             // this prevents some jumpiness when not necessary
             // and is particularly bad if you're in the middle of editing a comment
             return !(_.chain(self.case_uploads()).pluck('upload_id').isEqual(_(data).pluck('upload_id')).value() &&
-                _.chain(self.case_uploads()).pluck('task_status').isEqual(_(data).pluck('task_status')).value());
+                _.chain(self.case_uploads()).pluck('task_status').map(function (task_status) {return task_status()})
+                    .isEqual(_(data).pluck('task_status').map(function (task_status) {return task_status()})).value());
```

Comment:
This is a pretty gnarly statement to parse. When I have a big boolean statement like this, I sometimes like to break it up into smaller parts and use variable names to help the reader understand what's going on. For instance, you could do something like this:
```js
var shouldUpdate = function (data) {
	var uploadIdsAreEqual = _.chain(self.case_uploads())
		.pluck('upload_id')
		.isEqual(_(data).pluck('upload_id'))
		.value()
	var newTaskStatuses = _(data).pluck('task_status').map(function (task_status) {
		return task_status()
	})
	var taskStatusesAreEqual = _.chain(self.case_uploads())
		.pluck('task_status')
		.map(function (task_status) {
			return task_status()
		})
		.isEqual(newTaskStatuses)
		.value());
	return !(uploadIdsAreEqual && taskStatusesAreEqual);
```
It's much easier to read `return !(uploadIdsAreEqual && taskStatusesAreEqual);` than it is the whole statement at once.